agreement/abstractions.go

```diff
@@ -225,6 +225,10 @@ type KeyManager interface {
 	// valid for the provided votingRound, and were available at
 	// keysRound.
 	VotingKeys(votingRound, keysRound basics.Round) []account.Participation
+
+	// RecordAsync indicates that the given participation action has been taken.
+	// The operation needs to be asynchronous to avoid impacting agreement.
+	RecordAsync(account basics.Address, round basics.Round, participationType account.ParticipationAction)
```

Comment:
Could you rename this method to `RecordAsync` -> `RecordVote` ?

I think that the notation is "Async" should not be placed here, but rather on the implementation side.